willow/plugins/pillow.py

```diff
@@ -158,13 +183,15 @@ def save_as_jpeg(self, f, quality=85, optimize=False, progressive=False):
             image = self.image
 
         # Pillow only checks presence of optimize kwarg, not its value
-        kwargs = {}
+        kwargs = {
+            "quality": quality
+        }
         if optimize:
             kwargs['optimize'] = True
         if progressive:
             kwargs['progressive'] = True
 
-        image.save(f, 'JPEG', quality=quality, **kwargs)
+        f = self.save_and_optimize(f, image, "JPEG", **kwargs)
```

Comment:
Nitpick: We could retain the simplicity of the original link by removing `f = ` here, since `f` is written to directly by `save_and_optimize()`